### PR TITLE
Navbar skeleton

### PR DESCRIPTION
**Pull request by iaheck**
includes hook and skeletons for navbar curriculums
pending: skeleton in mobile don't work that well. but scopehammering

### CODE REVIEW

File: hooks/use-curriculums.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import useFetch from "./use-fetch";
+
+const useCurriculums = () => {
+  const { data, error } = useFetch(`/api/curriculums`);
+
+  if (error || !data) return undefined;
+  const curriculums = data
+    ?.map((curriculum) => ({
```
Comment (rtichauerv) [inline hooks/use-curriculums.js:8]: Is the `?` necessary if earlier you are returning another result when data is falsey? I would think it isn't.

**Review by rtichauerv — CHANGES_REQUESTED** [review]
I don't think it is a good practice to let the children components handle the request and decide wether they should be a skeleton or a list, I think the good way to do this is to handle this logic in the parent component, and for it to decide wether to render a skeleton or the menu. 
In other words, a Menu component should render its options, it's not its job to determine if it has the options, and what to render in different cases.

File: components/navbar/MenuManager.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Skeleton } from "@mui/material";
+import DesktopMenu from "./DesktopMenu";
+import MobileMenu from "./MobileMenu";
+
+export default function MenuManager({ version, loggedIn, pages }) {
```
Comment (ristofer) [inline components/navbar/MenuManager.js:5]: It is better to do the export default at the end of file?

File: components/navbar/MenuManager.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  if (version === "desktop") {
```
Comment (ristofer) [inline components/navbar/MenuManager.js:6]: Usually this is done through media queries on css.  I think is acceptable here.

File: components/navbar/Navbar.js [hunk 4]
```diff
@@ -26,13 +17,21 @@ function Navbar() {
             variant="h6"
             flexGrow={0}
           />
-          <MobileMenu pages={pages} loggedIn={!!currentUser} />
+          <MenuManager
+            version="mobile"
+            loggedIn={!!currentUser}
```
Comment (ristofer) [inline components/navbar/Navbar.js:22]: Could be better to use Boolean(currentUser) ?

File: hooks/use-curriculums.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const useCurriculums = () => {
+  const { data, error } = useFetch(`/api/curriculums`);
+
+  if (error || !data) return undefined;
+  const curriculums = data
+    .map((curriculum) => ({
+      text: curriculum.name,
+      url: `/curriculums/${curriculum.id}`,
+    }))
+    .slice(0, 3);
```
Comment (ristofer) [inline hooks/use-curriculums.js:12]: I think that .slice(0, 3); it is too obscure. Could be better to define a constant with a name.  Like "numberOfCurriculums"?

**Review by ristofer — CHANGES_REQUESTED** [review]
Just the hooks/use-curriculums.js need a change to understand better what .slice(0,3) is doing.

File: hooks/use-curriculums.js (hunk 5 above)
Comment (rtichauerv) [reply hooks/use-curriculums.js:12]: Agree, done

File: components/navbar/Navbar.js (hunk 4 above)
Comment (rtichauerv) [reply components/navbar/Navbar.js:22]: Good idea, didn't know that method

File: components/navbar/MenuManager.js (hunk 2 above)
Comment (rtichauerv) [reply components/navbar/MenuManager.js:5]: Done

**Review by ristofer — APPROVED** [review]
All changes were made :)

**Review by ristofer — APPROVED** [review]
All changes were made :)

**Review by rtichauerv — APPROVED** [review]

**Pull request merged**